src/app/clusters/application-launcher-server/application-launcher-server.cpp

```diff
@@ -162,10 +202,73 @@ bool emberAfApplicationLauncherClusterLaunchAppRequestCallback(app::CommandHandl
     auto & data         = commandData.data;
     auto & application  = commandData.application;
 
+    ApplicationLauncherApplication appInput;
+    appInput.catalogVendorId = application.catalogVendorId;
+    appInput.applicationId   = (application.applicationId.size() > 0) ? application.applicationId : data;
+
     Delegate * delegate = GetDelegate(endpoint);
     VerifyOrExit(isDelegateNull(delegate, endpoint) != true, err = CHIP_ERROR_INCORRECT_STATE);
 
     {
+#if CHIP_DEVICE_CONFIG_APP_PLATFORM_ENABLED
+        // if the feature flag is APP_Plaform, then this is a call to endpoint 1 to launching an app
+        //  1. Find target Content App and load if not loaded
+        //  2. Set current app to Content App
+        //  3. Set Content App status (basic cluster) to ACTIVE_VISIBLE_FOCUS
+        //  4. Call launch app command on Content App
+        if (delegate->HasFeature(chip::app::Clusters::ApplicationLauncher::ApplicationLauncherFeature::kApplicationPlatform))
+        {
+            ChipLogError(Zcl, "ApplicationLauncher has content platform feature");
+            ContentApp * app = chip::AppPlatform::AppPlatform::GetInstance().GetLoadContentAppByAppId(appInput);
+            if (app == NULL)
+            {
+                ChipLogError(Zcl, "ApplicationLauncher target app not found");
+                Commands::LauncherResponse::Type response;
+                response.data   = chip::CharSpan("data", strlen("data"));
+                response.status = StatusEnum::kAppNotAvailable;
+                err             = command->AddResponseData(commandPath, response);
+                SuccessOrExit(err);
+            }
+            chip::app::Clusters::ApplicationBasic::Structs::ApplicationBasicApplication::Type applicationType =
```

Comment:
```suggestion
            ApplicationBasic::Structs::ApplicationBasicApplication::Type applicationType =
```
and similar throughout this function.

Reply:
fixed